File: libs/shared/src/schema/snapshot.ts
import { z } from "zod";

export const CountrySchema = z.object({
  tag: z.string().min(2).max(4),
  treasury: z.number().optional(),
  gdp: z.number().optional(),
  market: z.string().optional(),
});

export const SnapshotSchema = z.object({
  channelId: z.string().min(1),
  saveHash: z.string().min(1),
  seq: z.number().min(0),
  countries: z.array(CountrySchema).max(300),
  updatedAt: z.string().datetime().optional(),
});

export type Country = z.infer<typeof CountrySchema>;
export type Snapshot = z.infer<typeof SnapshotSchema>;

File: apps/desktop/src/Vic3Unofficial.Twitch.Desktop/Models/Snapshot.cs
using System;
using System.Collections.Generic;

namespace Vic3Unofficial.Twitch.Desktop.Models;

public sealed class Snapshot
{
    public string ChannelId { get; set; } = "";
    public string SaveHash { get; set; } = "";
    public int Seq { get; set; }
    public List<Country> Countries { get; set; } = new();
    public string UpdatedAt { get; set; } = DateTime.UtcNow.ToString("o");
}
